### ISSUE TITLE
The `timeout` parameter in the client-level config is not being used

### ISSUE THREAD

**Issue by viacheslav-rostovtsev** [googleapis/gapic-generator-ruby#650]
For any given GAPIC client it is possible to set a `timeout` parameter in that client's `config`. E.g. for speech_v1:
```
     client = ::Google::Cloud::Speech::V1::Speech::Client.new do |config|
       config.timeout = 10.0
      end
```

However this `timeout` parameter is not being used for setting up grpc channel or being applied when individual RPCs are called.

**Comment by viacheslav-rostovtsev** [comment]
see also:
- an example PR fixing this: https://github.com/googleapis/gapic-generator-ruby/pull/639
- https://github.com/googleapis/gapic-generator-ruby/issues/192 where the purpose of the client-level timeout was discussed
- https://github.com/googleapis/gapic-generator-ruby/pull/209 the PR that introduced the issue following the discussion in https://github.com/googleapis/gapic-generator-ruby/issues/192

**Comment by viacheslav-rostovtsev** [comment]
Closed by #639
